kore/src/Kore/Step/Simplification/TermLike.hs

```diff
@@ -369,64 +369,83 @@ simplifyInternal term predicate = do
             --
             AndF andF ->
                 And.simplify =<< simplifyChildren andF
-            ApplySymbolF applySymbolF ->
-                Application.simplify predicate
+            ApplySymbolF applySymbolF -> OrPattern.fromPattern <$>
```

Comment:
Why are we unpacking the `Pattern` we just packed up?

Reply:
Strictly speaking, this is not unpacking, it's wrapping: this is an `OrPattern` packed in a `Pattern` wrapped in an `OrPattern`. I am doing the `OrPattern -> Pattern` packing so that the `or` simplifier has its chance to simplify the `or`, and I am doing the `Pattern -> OrPattern` wrapping because `OrPattern` is still the standard data structure in `TermLike.simplify`.

This seems somewhat ridiculous, so I'll try to get rid of it (either by returning `OrPattern`s or by making `Pattern` the standard thing in `TermLike.simplify`), but I think that the `Pattern -> OrPattern` wrapping is not that bad for performance.